scripts/patrol/patrol.py

```diff
@@ -793,64 +792,55 @@ def balance_hunting(self, possible_patrols: list):
-                needed_tags.append(f"{prey_type}")
-                if prey_type != "small_prey":
-                    needed_tags.append(f"{prey_type}0")
-                    needed_tags.append(f"{prey_type}1")
-                    needed_tags.append(f"{prey_type}2")
-                    needed_tags.append(f"{prey_type}3")
-        elif chosen_tag == "more_prey":
-            not_allowed_tag = "death"
-            needed_tags.append("large_prey")
-            needed_tags.append("huge_prey")
-            needed_tags.append("huge_prey0")
-            needed_tags.append("huge_prey1")
-            needed_tags.append("huge_prey2")
-            needed_tags.append("huge_prey3")
-
-        # filter all possible patrol depending on the needed tags
-        # one of the mentioned tags should be in the patrol tag
+        # get first what kind of prey size which will be chosen
+        biome = game.clan.biome
+        season = game.clan.current_season
+        possible_prey_size = []
+        idx = 0
+        prey_size = ["very_small", "small", "medium", "large", "huge"]
+        for amount in PATROL_BALANCE[biome][season]:
+            possible_prey_size.extend(repeat(prey_size[idx],amount))
+            idx += 1
+        chosen_prey_size = choice(possible_prey_size)
+        print(f"chosen filter prey size: {chosen_prey_size}")
+
+        # filter all possible patrol depending on the needed prey size
         for patrol in possible_patrols:
-            for tag in needed_tags:
-                if tag in patrol.tags:
-                    # if there is a tag set, check if this tag is not in the current patrol
-                    if not_allowed_tag and not_allowed_tag not in patrol.tags:
-                        filtered_patrols.append(patrol)
-                        break
-                    # when there is no tag set, add the patrol
-                    elif not not_allowed_tag:
-                        filtered_patrols.append(patrol)
-                        break
+            for adaption, needed_weight in PATROL_WEIGHT_ADAPTION.items():
+                if patrol.weight in range(needed_weight[0],needed_weight[1]):
+                    # get the amount of class sizes which can be increased
+                    increment = int(adaption.split("_")[0])
+                    new_idx = prey_size.index(chosen_prey_size) + increment
+                    # check that the increment does not lead to a overflow
+                    new_idx = new_idx if new_idx <= len(chosen_prey_size) else len(chosen_prey_size)
+                    chosen_prey_size = prey_size[new_idx]
+                    break
+
+            # now count the outcomes + prey size
+            prey_types = {}
+            for outcome in patrol.success_outcomes:
+                # ignore skill or trait outcomes
+                if outcome.stat_trait or outcome.stat_skill:
+                    continue
+                if outcome.prey:
+                    if outcome.prey[0] in prey_types:
```

Comment:
Will this break if more than one prey tag is ever used? At the moment, that isn't an issue, since there are only prey tags related to amount. But if other prey-tags are ever added, I imagine that might cause an issue on the balancing front.

Reply:
I think this should break, if a new prey tag is added and the balancing dictionary + filtering is not updated. Otherwise, it might slip your eyes, and we can't find the issue of prey balancing. I'm not sure what tags you did have in mind, but when they change the amount of prey which can get caught, it should be added to the according dictionaries, otherwise balancing wouldn't make any sense.

Reply:
I think this is OK for now, but we might consider rethinking the patrol-picking procedure to allow balancing to be built-in. I have a few idea, and I might start testing them out soon.